Introduction to PHP Operators (#2895)
section-engineering-education/engineering-education: content/articles/introduction-to-php-operators/index.md

### Introduction
An operator is is a symbol that performs operations on operands. In [PHP(Hypertext Preprocessor)](https://www.php.net/), there are different types of operators depending on the number of operands they take. For example, urinary operators take only one operand. On the contrary, binary operators take two operands..

> ### Prerequisites
> 
> 1.  Xampp for `Linux` downloaded on your machine. You can download the latest version [here](https://www.apachefriends.org/download.html) or follow [this]() steps on how to download.

Reviewer: ```suggestion
### Prerequisites
1.  Xampp for `Linux` downloaded on your machine. You can download the latest version [here](https://www.apachefriends.org/download.html) or follow [this]() steps on how to download.
```

2. Little knowledge of PHP.

> 3.  A text editor like [Visual studio](https://code.visualstudio.com/download) or any other depending on your taste.
> 
> `NB`
> To run any PHP file, You must first start `Xampp`. This is done using this command.
> ```bash

Reviewer: ```suggestion
3.  A text editor like [Visual studio](https://code.visualstudio.com/download) or any other depending on your taste.
>To run any PHP file, You must first start `Xampp`. This is done using this command.
```bash
```

   $num=10+20;`
 ```  

  In the above example  `+` is the `operator` while `10` and `20` are `Operands`

 An  `Operator` is a symbol used to perform operations on operands/variables. 

 `Operand` is the object being acted upon. 10 and 20 are being added in the above example. 

 NB. `$(dollar)` sigh is used to create a variable `num` which holds the values 10 and 20.

 #### Creating a php file in Xampp

 Now that we know what an operator and an operand are, let us start working with them. To do that we have to create a `PHP file` in Xampp. 

 After downloading xampp, you will notice that a folder `/opt/lampp/htdocs` will be created. Open  `/opt/lampp/htdocs` using terminal to create a file. To open, type:

 ```bash
 $ cd /opt/lampp/htdocs
 ```
 Let us create a folder `operators` in `htdocs` which we will use to store all our files.

Type the command below to create:
```bash
$ sudo mkdir operators
```
To create a php file in `/opt/lampp/htdocs/operators`, type:

```bash
$ cd /opt/lampp/htdocs/operators 
$ sudo nano touch example.php
```
The above command will create a PHP file called `example.php` in the operators' folder. We will use this file later. 

#### Arithmetic Operators

> These are operators that are used on the operator to execute arithmetic operations.
> 
> There are various Arithmetic operators supported by PHP:
> 
> 1. Addition Operator(+)  - They add two operands.
> 
> 2. Subtraction Operator(-) - Subtract the second operand from the first operand.
> 
> 3. Muitiplacation Operators (*) - These multiply two operands.
> 
> 4. Division Operator (/) - Divide numerator by de-numerator
> 

Reviewer: ```suggestion
These are operators that are used on the operator to execute arithmetic operations.

There are various Arithmetic operators supported by PHP:

1. Addition Operator(+)  - It adds two operands.

2. Subtraction Operator(-) - Subtracts the second operand from the first operand.

3. Muitiplacation Operators (*) - These multiplies two operands.

4. Division Operator (/) - Divides numerator by denominator

```

6. Less than (<)- If the operand on the left is smaller than the operand on the right, return true; otherwise, return false. -->

7. Less than or equal(<=) - If the left operand is smaller than or equal to the right operand, return true; otherwise, return false. 

Lets have a look at Program that illustrate the working of a Comparison operators:
```php
<?php
$x = 100;  
$y = 100;
var_dump($x == $y) ; // returns true because values are equal . ""
echo"<br>";
var_dump($x === $y).'<br/>'; // returns false because types are not equal
echo"<br>";
var_dump($x != $y); // returns false because values are equal
echo"<br>";
var_dump($x !== $y); // returns true because types are not equal
echo"<br>";
var_dump($x > $y); // returns true because $x is greater than $y
echo"<br>";
var_dump($x <= $y); // returns true because $x is less than or equal to $y
echo"<br>";
var_dump($x < $y); // returns true because $x is less than $y
echo"<br>";
var_dump($x <= $y); // returns true because $x is less than or equal to $y
?>
```
Output
```
bool(true)
bool(true)
bool(false)
bool(false)
bool(false)
bool(true)
bool(false)
bool(true)  
```
#### String Operators
In PHP we have two string operators:

> . `concatenation operator` ('.') -Returns the right and left arguments concatenated.
> 
> . `concatenation assignment operator`  ('.=') -  Appends the right-hand argument to the left-hand argument.
> 
> A. Following `concatenation operator` we have the following Example:
> ```php
> $a = "Hello ";
> $b = $a . "World!"; // here, $b contains "Hello World!"
> echo "$b";
> ?>
> ```
> Output
> ```
> Hello World! 
> ```
> B. Following `concatenation assignment operator` we have the following Example:
> ```php
> $a = "Good ";
> $a .= "Work!"; // here, $a contains "Good Work!"
> echo "$a";
> ?>
> ```

Reviewer: ```suggestion
The `concatenation operator` ('.')  which returns the right and left arguments concatenated and the  `concatenation assignment operator`  ('.=')  which appends the right-hand argument to the left-hand argument.

- Concatenation Operator.  The example below demonstrates the use of a `concatenation operator` :
```php
$a = "Hello ";
$b = $a . "World!"; // here, $b contains "Hello World!"
echo "$b";
?>
```
Output
```
Hello World! 
```
- Concatenation Assignment Operator.  The example below demonstrates the use of a `concatenation assignment operator` :
```php
$a = "Good ";
$a .= "Work!"; // here, $a contains "Good Work!"
echo "$a";
?>
```
```
